lib/discourse_api/api/categories.rb

```diff
@@ -75,12 +75,17 @@ def category(id)
         response[:body]['category']
       end
 
+      # TODO: Deprecated. Remove after 20210727
       def category_set_user_notification(args = {})
         category_id = args[:id]
         args = API.params(args)
           .required(:notification_level)
         post("/category/#{category_id}/notifications", args)
       end
+
+      def category_set_user_notification_level(category_id, notification_level)
```

Comment:
We still want the 2nd param to be a hash.

```
def category_set_user_notification_level(category_id, params = {})
args = API.params(params)
  .required(:notification_level)
```
This way we can add more parameters to the hash if they are added in the discourse code base without having to change this method signature.